src/Common/src/System/Security/Cryptography/RSACng.ImportExport.cs

```diff
@@ -0,0 +1,225 @@
+                    pBcryptBlob->cbPublicExp = parameters.Exponent.Length;
+                    pBcryptBlob->cbModulus = parameters.Modulus.Length;
+
+                    if (includePrivate)
+                    {
+                        pBcryptBlob->cbPrime1 = parameters.P.Length;
+                        pBcryptBlob->cbPrime2 = parameters.Q.Length;
+                    }
+
+                    int offset = sizeof(BCRYPT_RSAKEY_BLOB);
+
+                    Emit(rsaBlob, ref offset, parameters.Exponent);
+                    Emit(rsaBlob, ref offset, parameters.Modulus);
+
+                    if (includePrivate)
+                    {
+                        Emit(rsaBlob, ref offset, parameters.P);
+                        Emit(rsaBlob, ref offset, parameters.Q);
+                    }
+
+                    // We better have computed the right allocation size above!
+                    Debug.Assert(offset == blobSize, "offset == blobSize");
+                }
+
+                ImportKeyBlob(rsaBlob, includePrivate);
+            }
+        }
+
+        /// <summary>
+        ///     Append "value" to the data already in "rsaBlob".
+        /// </summary>
+        private static void Emit(byte[] rsaBlob, ref int offset, byte[] value)
+        {
+            Buffer.BlockCopy(value, 0, rsaBlob, offset, value.Length);
+            offset += value.Length;
+        }
+
+        /// <summary>
+        ///     Exports the key used by the RSA object into an RSAParameters object.
+        /// </summary>
+        public override RSAParameters ExportParameters(bool includePrivateParameters)
+        {
+            byte[] rsaBlob = ExportKeyBlob(includePrivateParameters);
+            RSAParameters rsaParams = new RSAParameters();
+            ExportParameters(ref rsaParams, rsaBlob, includePrivateParameters);
+            return rsaParams;
+        }
+
+        private static void ExportParameters(ref RSAParameters rsaParams, byte[] rsaBlob, bool includePrivateParameters)
+        {
+            //
+            // We now have a buffer laid out as follows:
+            //     BCRYPT_RSAKEY_BLOB   header
+            //     byte[cbPublicExp]    publicExponent      - Exponent
+            //     byte[cbModulus]      modulus             - Modulus
+            //     -- Private only --
+            //     byte[cbPrime1]       prime1              - P
+            //     byte[cbPrime2]       prime2              - Q
+            //     byte[cbPrime1]       exponent1           - DP
+            //     byte[cbPrime2]       exponent2           - DQ
+            //     byte[cbPrime1]       coefficient         - InverseQ
+            //     byte[cbModulus]      privateExponent     - D
+            //
+            byte[] tempMagic = new byte[4];
+            tempMagic[0] = rsaBlob[0]; tempMagic[1] = rsaBlob[1]; tempMagic[2] = rsaBlob[2]; tempMagic[3] = rsaBlob[3];
+            KeyBlobMagicNumber magic = (KeyBlobMagicNumber)BitConverter.ToInt32(tempMagic, 0);
```

Comment:
Could the above be simplified to avoid the `byte[] tempMagic` allocation?

```csharp
KeyBlobMagicNumber magic = (KeyBlobMagicNumber)BitConverter.ToInt32(rsaBlob, 0);
```